crates/dht/src/arc_set.rs

```diff
@@ -0,0 +1,278 @@
+//! Represents a set of [DhtArc]s.
+//!
+//! A set of [DhtArc]s is combined as a set union into an [ArcSet].
+//!
+//! To restrict [crate::dht::Dht] operations to a specific set of sectors, the [ArcSet]s of two
+//! DHTs can be intersected to find the common sectors, using [ArcSet::intersection].
+
+use kitsune2_api::{DhtArc, K2Error, K2Result};
+use std::collections::HashSet;
+
+/// Represents a set of [DhtArc]s.
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct ArcSet {
+    inner: HashSet<u32>,
```

Comment:
hehe : )

Reply:
Yes it's a little odd, but it's using sector id values so it's size limited to the number of sectors we chose which is 512 and having set operations makes life a little easier... I could probably simplify the constructor thinking about it actually. I'll leave that alone for now

Reply:
All good. It's perfect for a first implementation. It totally may never need to be optimized further.